dashboard/app/controllers/api/v1/sections_controller.rb

```diff
@@ -119,13 +123,25 @@ def join
       render_404
       return
     end
+    # If user exceeded 3 section attempts in 24 hours, check for captcha verification
+    if current_user.display_captcha?
+      recaptcha_token = params[:recaptchaToken] || ""
+      is_verified = verify_recaptcha_token(recaptcha_token)
+      current_user.increment_section_attempts
+      unless is_verified
+        render json: {
+          result: 'captcha_failed'
+        }, status: :bad_request
+      end
+    end
+
     result = @section.add_student current_user
+    current_user.increment_section_attempts
```

Comment:
Are this call and the one on line 130 mutually exclusive?  It looks like it's possible to call `increment_section_attempts` in both places which would double-count a single attempt?

Reply:
I put some debugger statements in to test this out, and it does not seem to be double counting. I think the `load_resource` hook comes from the CanCan gem, and if it can't load the resource successfully (ie, can't create the instance variable `@section`), then the code in `join` will never be hit.